cli/tools/registry/module_content.rs

```diff
@@ -317,12 +317,23 @@ mod test {
     }"#,
         None,
       ),
-      ("/package-b/deno.json", "{}", None),
+      ("/package-b/deno.json", r#"{
+        "compilerOptions": { "jsx": "react-jsx" },
+        "imports": {
+          "react": "npm:react"
+          "@types/react": "npm:@types/react"
+        }
+      }"#, None),
       (
         "/package-a/main.tsx",
         "export const component = <div></div>;",
         Some("/** @jsxRuntime automatic *//** @jsxImportSource npm:react *//** @jsxImportSourceTypes npm:@types/react *//** @jsxFactory React.createElement *//** @jsxFragmentFactory React.Fragment */export const component = <div></div>;"),
       ),
+      (
+        "/package-b/main.tsx",
+        "export const componentB = <div></div>;",
+        Some("/** @jsxRuntime automatic *//** @jsxImportSource npm:react *//** @jsxImportSourceTypes npm:react *//** @jsxFactory React.createElement *//** @jsxFragmentFactory React.Fragment */export const componentB = <div></div>;"),
```

Comment:
I just realized this is an interesting edge case where at runtime it would actually resolve to `npm:@types/react` instead of `npm:react`... so maybe this is not exactly correct. This is at least better than before where these wouldn't be inserted.

Reply:
Hmmm, IIRC @nathanwhit had a very similar problem in the LSP where the imports were pointing to `npm:@types/react` instead of `npm:react` - maybe he could clear up this situation?

Reply:
I though of a solution and opened https://github.com/denoland/deno/issues/28231 which should fix this. This behaviour is already broken so I think this PR is good to merge.